protocol/README.md

```diff
@@ -1,7 +1,36 @@
-## Ably Realtime Error Codes
+# Ably Realtime Error Codes
+
+## Overview
 
 Ably's [client library SDKs](https://www.ably.io/download), and the [core realtime platform](https://www.ably.io/platform), use common error codes. These error codes are stored and maintained in the shared public JSON file [errors.json](./errors.json).
 
 In addition, a list of error codes and corresponding support articles to help understand how to resolve common problems relating to these error codes can be seen in [errorsHelp.json](./errorsHelp.json).
 
 If you need help understanding any error codes, or need technical support, please visit the [Ably support desk](https://www.ably.io/support).
+
+## Error Code Ranges
+
+Codes in [error.json](error.json) fall into these ranges:
+
+| From  | Title |
+| ----- | ----- |
+| 10000 | Generic |
+| 40000 | 400 |
+| 40100 | 401 |
+| 40300 | 403 |
+| 40400 | 404 |
+| 40500 | 405 |
+| 41001 | 410 |
+| 42200 | 422 |
+| 42910 | 429 |
+| 50000 | 500 |
+| 50010 | Edge cacheing / proxy service error codes |
+| 50320 | Active Traffic Management error code to indicate intentional redirect of traffic to fallback hosts |
```

Comment:
Not sure that this categorization really fits how the codes are used right now. E.g. 'edge caching' doesn't consume the whole block of three hundred codes from 50010 to 50310 - rather edge caching codes are a few normal http status codes with "10" added to the end. The router uses a similar thing but with '05', so 50205/50305/50405. The idea's was to have ties between the codes and the http status codes - have them be status codes with 'extra detail'. (though we don't do that consistently).

(I realise this isn't a new problem with this pr, the json comments gave a similar impression)

No strong opinion on whether this is actually a problem and if so what the solution is, happy for Quintin to say whether/how he'd like to redesign any of this, whether in the direction of stronger ties to http status codes, or weaker, or whatever

Reply:
I considered adding a "To" column but I would have been plucking those out of thin air. Hence I went with this lighter weight approach, directly reflecting what was in the existing JSON comments, albeit that only works if they're in order!

Perhaps the next iteration, ideally beyond the scope of this PR, would be to add that To column.

Reply:
https://github.com/ably/ably-common/issues/76 👈  @SimonWoolf, if you're happy with that deferment then could you please resolve this conversation?

Reply:
I'm going to resolve this conversation on your behalf, @SimonWoolf, as a prerequisite to landing the PR. If you need to follow this up further then please do so under #76. Thanks.